ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -64,13 +67,15 @@
   static final private String CLASS_NAME = Cleaner.class.getName();
   static final private Logger LOG = LoggerFactory.getLogger(CLASS_NAME);
   private long cleanerCheckInterval = 0;
+  private Executor cleanerExecutor;
 
   private ReplChangeManager replChangeManager;
 
   @Override
   public void init(AtomicBoolean stop) throws Exception {
     super.init(stop);
     replChangeManager = ReplChangeManager.getInstance(conf);
+    cleanerExecutor = Executors.newFixedThreadPool(conf.getIntVar(HiveConf.ConfVars.HIVE_COMPACTOR_CLEANER_REQUEST_QUEUE));
```

Comment:
I think it would be a good idea to shut down the executor when we finished the run loop.
What do you think?

Reply:
Yeah, makes sense. Added it.

Reply:
I prefer the way how it was done in Worker.java:
- Named threads
- Priority down
- Set daemon on/off
- Stopped in finally
Shall we do it here, or should we create a follow-up jira for creating and cleaning up the executor threads for Cleaner and Initiator as well?

Thanks,
Peter

Reply:
Just saw the approach in Worker.java, it looks much better/cleaner. I will make the changes here only ()for both cleaner/intiator) and update once it is ready.

Reply:
Done.